### PR TITLE
[ja] Translate tasks/debug-application-cluster/resource-metrics-pipeline into Japanese

### LINKED ISSUE #30871: [ja] Translate tasks/debug-application-cluster/resource-metrics-pipeline into Japanese
# This is a Feature Request

## What would you like to be added

Translate [tasks/debug-application-cluster/resource-metrics-pipeline](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-metrics-pipeline/)  into Japanese.

## Why is this needed

This page has not been translated.

## Comments

/language ja
/assign

### PR DESCRIPTION
**Pull request by ptux**
resolve #30871

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/assign @atoato88

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+reviewers:
+- ptux
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:3]: please remove reviewers block

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+title: リソースメトリクスパイプライン
+content_type: concept
+---
+
+<!-- overview -->
+
+Kubernetesでは、コンテナのCPU使用率やメモリ使用率といったリソース使用量のメトリクスが、メトリクスAPIを通じて提供されています。これらのメトリクスは、ユーザーが`kubectl top`コマンドで直接アクセスするか、クラスタ内のコントローラ（例えばHorizontal Pod Autoscaler）が判断するためにアクセスすることができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:10]:
```suggestion
Kubernetesでは、コンテナのCPU使用率やメモリ使用率といったリソース使用量のメトリクスが、メトリクスAPIを通じて提供されています。これらのメトリクスは、ユーザーが`kubectl top`コマンドで直接アクセスするか、クラスター内のコントローラー(例えばHorizontal Pod Autoscaler)が判断するためにアクセスすることができます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+<!-- body -->
+
+## メトリクスAPI
+
+メトリクスAPIを使用すると、指定したノードやポッドが現在使用しているリソース量を取得することができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:16]:
```suggestion
メトリクスAPIを使用すると、指定したノードやPodが現在使用しているリソース量を取得することができます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+このAPIはメトリックの値を保存しないので、例えば10分前に指定されたノードが使用したリソース量を取得することはできません。
+
+メトリクスAPIは他のAPIと何ら変わりはありません。
+
+- 他のKubernetes APIと同じエンドポイントを経由して、`/apis/metrics.k8s.io/`パスの下で発見できます。
+- 同じセキュリティ、スケーラビリティ、信頼性の保証を提供します。
+
+メトリクスAPIは[k8s.io/metrics](https://github.com/kubernetes/metrics/blob/master/pkg/apis/metrics/v1beta1/types.go)レポジトリで定義されています。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:24]:
```suggestion
メトリクスAPIは[k8s.io/metrics](https://github.com/kubernetes/metrics/blob/master/pkg/apis/metrics/v1beta1/types.go)リポジトリで定義されています。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+メトリクスAPIについての詳しい情報はそちらをご覧ください。
+
+{{< note >}}
+メトリクスAPIを使用するには、クラスタ内にメトリクスサーバが配置されている必要があります。そうでない場合は利用できません。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:28]:
```suggestion
メトリクスAPIを使用するには、クラスタ内にメトリクスサーバーが配置されている必要があります。そうでない場合は利用できません。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+{{< /note >}}
+
+## リソース使用量の測定
+
+### CPU
+
+CPU は、一定期間の平均使用量を [CPU cores](/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu) という単位で報告されます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:35]:
```suggestion
CPUは、一定期間の平均使用量を[CPU cores](/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)という単位で報告されます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+この値は、カーネルが提供する累積CPUカウンターをレートすることで得られます（LinuxとWindowsの両カーネルで）。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:36]:
```suggestion
この値は、カーネルが提供する累積CPUカウンターをレートすることで得られます(LinuxとWindowsの両カーネルで)。
```

rateも訳したいですね。

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+kubeletは、レート計算のためのウィンドウを選択します。
+
+### メモリ
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:39]:
```suggestion
### メモリー
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+メモリは、測定値が収集された時点のワーキングセットとして、バイト単位で報告されます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:41]:
```suggestion
メモリーは、測定値が収集された時点のワーキングセットとして、バイト単位で報告されます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+理想的な世界では、「ワーキングセット」は、メモリ不足で解放できない使用中のメモリ量です。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:42]:
```suggestion
理想的な世界では、「ワーキングセット」は、メモリー不足で解放できない使用中のメモリー量です。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+しかし、ワーキングセットの計算はホストOSによって異なり、一般に推定値を生成するためにヒューリスティックを多用しています。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:43]:
```suggestion
しかし、ワーキングセットの計算はホストOSによって異なり、一般に推定値を生成するために経験則を多用しています。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+Kubernetesはスワップをサポートしていないため、すべての匿名（非ファイルバックアップ）メモリが含まれます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:44]:
```suggestion
Kubernetesはスワップをサポートしていないため、すべての匿名(非ファイルバックアップ)メモリーが含まれます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+ホストOSは常にそのようなページを再請求することができないため、メトリックには通常、一部のキャッシュされた（ファイルバックされた）メモリも含まれます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:45]:
```suggestion
ホストOSは常にそのようなページを再請求することができないため、メトリックには通常、一部のキャッシュされた(ファイルバックされた)メモリーも含まれます。
```
ファイルバックされた、とは日本語ではあまり言わない気がしますが、何か良い訳ないですかね @inductor

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+## メトリクスサーバー
+
+[メトリクスサーバー](https://github.com/kubernetes-sigs/metrics-server)は、クラスタ全体のリソース使用量データのアグリゲータです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:49]:
```suggestion
[メトリクスサーバー](https://github.com/kubernetes-sigs/metrics-server)は、クラスター全体のリソース使用量データのアグリゲーターです。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+デフォルトでは、`kube-up.sh`スクリプトで作成されたクラスタにDeploymentオブジェクトとしてデプロイされます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:50]:
```suggestion
デフォルトでは、`kube-up.sh`スクリプトで作成されたクラスターにDeploymentオブジェクトとしてデプロイされます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+別のKubernetesセットアップ機構を使用する場合は、提供される[deployment components.yaml](https://github.com/kubernetes-sigs/metrics-server/releases)ファイルを使用してデプロイすることができます。
+メトリクスサーバーは、Summary APIからメトリクスを収集します。
+各ノードの[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)から[Kubernetes aggregator](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)経由でメインAPIサーバに登録されるようになっています。
+
+メトリクスサーバーについては、[the design doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/metrics-server.md)で詳しく解説しています。
+
+
+### Summary APIソース
+
+[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)は、ノード、ボリューム、ポッド、コンテナレベルの統計情報を収集し、[Summary API(https://github.com/kubernetes/kubernetes/blob/7d309e0104fedb57280b261e5677d919cb2a0e2d/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go)]で省略して消費者が読めるようにするものです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:60]:
```suggestion
[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)は、ノード、ボリューム、Pod、コンテナレベルの統計情報を収集し、[Summary API(https://github.com/kubernetes/kubernetes/blob/7d309e0104fedb57280b261e5677d919cb2a0e2d/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go)]で省略して消費者が読めるようにするものです。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+1.23以前は、これらのリソースは主に[cAdvisor(https://github.com/google/cadvisor)]から収集されていました。しかし、1.23では`PodAndContainerStatsFromCRI FeatureGate`の導入により、コンテナとポッドレベルの統計情報をCRI実装で収集することができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:62]:
```suggestion
1.23以前は、これらのリソースは主に[cAdvisor](https://github.com/google/cadvisor)から収集されていました。しかし、1.23では`PodAndContainerStatsFromCRI FeatureGate`の導入により、コンテナとPodレベルの統計情報をCRI実装で収集することができます。
```

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+注意: これはCRI実装によるサポートも必要です (containerd >= 1.6.0, CRI-O >= 1.23.0)。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:64]:
```suggestion
注意: これはCRI実装によるサポートも必要です(containerd >= 1.6.0, CRI-O >= 1.23.0)。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+各ノードの[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)から[Kubernetes aggregator](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)経由でメインAPIサーバに登録されるようになっています。
+
+メトリクスサーバーについては、[the design doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/metrics-server.md)で詳しく解説しています。
+
+
+### Summary APIソース
+
+[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)は、ノード、ボリューム、Pod、コンテナレベルの統計情報を収集し、[Summary API](https://github.com/kubernetes/kubernetes/blob/7d309e0104fedb57280b261e5677d919cb2a0e2d/staging/src/k8s.io/kubelet/pkg/apis/stats/v1alpha1/types.go)で省略して消費者が読めるようにするものです。
+
+1.23以前は、これらのリソースは主に[cAdvisor](https://github.com/google/cadvisor)から収集されていました。しかし、1.23では`PodAndContainerStatsFromCRI FeatureGate`の導入により、コンテナとPodレベルの統計情報をCRI実装で収集することができます。
```
Comment (atoato88) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:60]:
```suggestion
1.23以前は、これらのリソースは主に[cAdvisor](https://github.com/google/cadvisor)から収集されていました。しかし、1.23では`PodAndContainerStatsFromCRI`フィーチャーゲートの導入により、コンテナとPodレベルの統計情報をCRI実装で収集することができます。
```
We use "フィーチャーゲート" as a Japanese translation for "FeatureGate".
Example: [here](https://kubernetes.io/ja/docs/reference/command-line-tools-reference/feature-gates/)

**Comment by atoato88** [comment]
Thank you to quick update for my comment.
nasa9084 left some suggestions, could you check those?

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 20]
```diff
@@ -0,0 +1,62 @@
+---
+title: リソースメトリクスパイプライン
+content_type: concept
+---
+
+<!-- overview -->
+
+Kubernetesでは、コンテナのCPU使用率やメモリー使用率といったリソース使用量のメトリクスが、メトリクスAPIを通じて提供されています。これらのメトリクスは、ユーザーが`kubectl top`コマンドで直接アクセスするか、クラスター内のコントローラー(例えばHorizontal Pod Autoscaler)が判断するためにアクセスすることができます。
```
Comment (inductor) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:8]:
```suggestion
Kubernetesでは、コンテナのCPU使用率やメモリ使用率といったリソース使用量のメトリクスが、メトリクスAPIを通じて提供されています。これらのメトリクスは、ユーザーが`kubectl top`コマンドで直接アクセスするか、クラスター内のコントローラー(例えばHorizontal Pod Autoscaler)が判断するためにアクセスすることができます。
```
Ref. https://github.com/kubernetes/website/search?q=%E3%83%A1%E3%83%A2%E3%83%AA

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+
+## リソース使用量の測定
+
+### CPU
+
+CPUは、一定期間の平均使用量を[CPU cores](/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu)という単位で報告されます。
+この値は、カーネルが提供する累積CPUカウンターの比率を取得することで得られます(LinuxとWindowsの両カーネルで)。
+kubeletは、比率計算のためのウィンドウを選択します。
+
+### メモリー
```
Comment (inductor) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:37]:
```suggestion
### メモリ
```

Please check all parts in the document

File: content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md [hunk 22]
```diff
@@ -0,0 +1,62 @@
...
+
+## メトリクスサーバー
+
+[メトリクスサーバー](https://github.com/kubernetes-sigs/metrics-server)は、クラスター全体のリソース使用量データのアグリゲーターです。
+デフォルトでは、`kube-up.sh`スクリプトで作成されたクラスターにDeploymentオブジェクトとしてデプロイされます。
+別のKubernetesセットアップ機構を使用する場合は、提供される[deployment components.yaml](https://github.com/kubernetes-sigs/metrics-server/releases)ファイルを使用してデプロイすることができます。
+メトリクスサーバーは、Summary APIからメトリクスを収集します。
+各ノードの[Kubelet](/docs/reference/command-line-tools-reference/kubelet/)から[Kubernetes aggregator](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)経由でメインAPIサーバーに登録されるようになっています。
+
+メトリクスサーバーについては、[the design doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/metrics-server.md)で詳しく解説しています。
```
Comment (inductor) [inline content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md:53]: Translate "the design doc"? It's not a proper noun

Comment (ptux) [reply content/ja/docs/tasks/debug-application-cluster/resource-metrics-pipeline.md]: fixed.

**Comment by ptux** [comment]
@inductor

cloud you give an LGTM label?

**Pull request merged**